### ISSUE TITLE
Does `SetupAllProperties` actually create any setups, or does it not?

### ISSUE THREAD

**Issue by stakx** [moq/moq4#681]
Using Moq 4.9.0, given this:

```
public interface T
{
    T P { get; }
}

var mock = new Mock<T>();
mock.SetupAllProperties();
```

**`mock.Verify(m => m.P, Times.Once)`** will produce this error message:

```
Expected invocation on the mock once, but was 0 times: m => m.P
No setups configured.
No invocations performed.
```

**`mock.VerifyAll()`** will produce a contradictory error message:

```
The following setups on mock 'Mock<T:00000001>' were not matched:
T m => m.P
```

One error message says that there are no setups, while the other says that there is a setup. Which one is (or should be) correct?

**Comment by stakx** [comment]
In my opinion, `mock.Verify(…)` has it right: `SetupAllProperties` should not create any setups (meaning that `VerifyAll` should succeed). Despite its name, the intent behind `SetupAllProperties` is to conveniently "auto-implement" all properties such that they remember the values they're being set to. It's unlikely that someone is going to read (and write to) each and every property to make `VerifyAll` fail. Or in other words, `VerifyAll` is currently quite useless when a prior call to `SetupAllProperties` was made.

I'd love to get some input on this. What do you think?

**Comment by kzu** [comment]
Absolutely, you got it right @stakx: SetupAllProperties is just a way to get useful stubs.

**Comment by stakx** [comment]
Great, thanks. I'll then probably change the error messages & `VerifyAll` implementation so this works more seamlessly.

**Comment by stakx** [comment]
Due to the decision made in #681 and the corresponding change made to SetupAllProperties in #684, this issue will be fixed in Moq version >4.9.0.
